x-pack/osquerybeat/internal/osqueryd/client.go

```diff
@@ -18,23 +19,57 @@ import (
 const (
 	DefaultTimeout = 30 * time.Second
 
+	lruCacheSize = 1024
+
 	retryWait  = 200 * time.Millisecond
 	retryTimes = 10
 	logTag     = "osqueryd_cli"
 )
 
+type Cache interface {
+	Add(key, value interface{}) (evicted bool)
+	Get(key interface{}) (value interface{}, ok bool)
+	Resize(size int) (evicted int)
+}
+
 type Client struct {
-	cli *osquery.ExtensionManagerClient
+	cli   *osquery.ExtensionManagerClient
+	cache Cache
+	log   *logp.Logger
 }
 
-func NewClient(ctx context.Context, socketPath string, to time.Duration) (*Client, error) {
+type Option func(*Client)
+
+func NewClient(ctx context.Context, socketPath string, to time.Duration, opts ...Option) (*Client, error) {
 	cli, err := newClientWithRetries(ctx, socketPath, to)
 	if err != nil {
 		return nil, err
 	}
-	return &Client{
+	c := &Client{
 		cli: cli,
-	}, nil
+	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	if c.log == nil {
+		c.log = logp.NewLogger(logTag)
+	}
```

Comment:
nit; new code should not use `NewLogger`. The logger should be assumed to be a dependency that must be passed in.